python/ray/serve/_private/logging_utils.py

```diff
@@ -69,39 +76,36 @@ def format(self, record: logging.LogRecord) -> str:
                 The formatted log record in json format.
         """
         record_format = copy.deepcopy(self.component_log_fmt)
-        if SERVE_LOG_REQUEST_ID in record.__dict__:
-            record_format[SERVE_LOG_REQUEST_ID] = SERVE_LOG_RECORD_FORMAT[
+        record_attributes = copy.deepcopy(record.__dict__)
+        record_format[SERVE_LOG_LEVEL_NAME] = record.levelname
+        record_format[SERVE_LOG_TIME] = self.asctime_formatter.format(record)
+
+        if SERVE_LOG_REQUEST_ID in record_attributes:
+            record_format[SERVE_LOG_REQUEST_ID] = record_attributes[
                 SERVE_LOG_REQUEST_ID
             ]
-        if SERVE_LOG_ROUTE in record.__dict__:
-            record_format[SERVE_LOG_ROUTE] = SERVE_LOG_RECORD_FORMAT[SERVE_LOG_ROUTE]
+        if SERVE_LOG_ROUTE in record_attributes:
+            record_format[SERVE_LOG_ROUTE] = record_attributes[SERVE_LOG_ROUTE]
 
-        if SERVE_LOG_APPLICATION in record.__dict__:
-            record_format[SERVE_LOG_APPLICATION] = SERVE_LOG_RECORD_FORMAT[
+        if SERVE_LOG_APPLICATION in record_attributes:
+            record_format[SERVE_LOG_APPLICATION] = record_attributes[
                 SERVE_LOG_APPLICATION
             ]
 
-        message_formatter = logging.Formatter(
-            SERVE_LOG_RECORD_FORMAT[SERVE_LOG_MESSAGE]
-        )
-        record_format[SERVE_LOG_MESSAGE] = message_formatter.format(record)
+        record_format[SERVE_LOG_MESSAGE] = self.message_formatter.format(record)
 
-        if SERVE_LOG_EXTRA_FIELDS in record.__dict__:
-            if not isinstance(record.__dict__[SERVE_LOG_EXTRA_FIELDS], dict):
+        if SERVE_LOG_EXTRA_FIELDS in record_attributes:
+            if not isinstance(record_attributes[SERVE_LOG_EXTRA_FIELDS], dict):
                 raise ValueError(
                     f"Expected a dictionary passing into {SERVE_LOG_EXTRA_FIELDS}, "
-                    f"but got {type(record.__dict__[SERVE_LOG_EXTRA_FIELDS])}"
+                    f"but got {type(record_attributes[SERVE_LOG_EXTRA_FIELDS])}"
                 )
-            for k, v in record.__dict__[SERVE_LOG_EXTRA_FIELDS].items():
+            for k, v in record_attributes[SERVE_LOG_EXTRA_FIELDS].items():
                 if k in record_format:
                     raise KeyError(f"Found duplicated key in the log record: {k}")
                 record_format[k] = v
 
-        # create a formatter using the format string
-        formatter = logging.Formatter(json.dumps(record_format))
```

Comment:
This formatter call will be called recursively in some case (when `tqdm` is used). Changed this formatter to just get the attributes directly instead of relying on `logging.Formatter`